---
title: "bulk-rnaseq-gallagher"
output: html_document
date: "2025-01-07"
---
#I don't know what this is. is it needed??
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


Note:
I am doing the analysis on the R server from class. All of the files are uploaded there and it seems like it would have taken a long time to download all of the files I would have needed. I'll try again when I'm done there. This will just be a copy and paste of my code but I won't be able to run it from my local computer because the files are not there.

Before code gets put in here I will have made sure that it works on the R server though.


#QC 
```{r}
#run fastqc on the raw fastq files
fastqcr::fastqc(fq.dir = "~/data/bulk_RNA_seq/fastq_files/raw",
                qc.dir = "/home/willettb/bulk-rnaseq-gallagher/fastq_raw_qc",
                threads = 1,
                fastqc.path = "~/software/FastQC/fastqc")

#multiqc to check qc data
system(command = 'multiqc /home/willettb/bulk-rnaseq-gallagher/fastq_raw_qc --filename *fastqc.html --outdir /home/willettb/bulk-rnaseq-gallagher')

```

#trimming - automated, but with a R-to-terminal problem
```{r}
#worked. but only one two files (one sample) Would like to automate this...
system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep1_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep1_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep1_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep1_2.fastq.gz')


#automating - I'm able to get all of the file names constructed correctly through this so they can be fed into cutadapt
#PROBLEM: running cutadapt through the command line doesn't let me use the R variables. Not sure how to export them
samples <- read.table("/home/willettb/bulk-rnaseq-gallagher/samples.txt",sep=" ",header=FALSE)

for (x in 1:3) {
file_output_1 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/", samples[1,1],"_trimmed.fastq.gz", sep = "")
file_input_1 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_raw/", samples[1,1], ".fastq.gz", sep = "")
print(file_output_1)
print(file_input_1)
file_output_2 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/", samples[1,2],"_trimmed.fastq.gz", sep = "")
file_input_2 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_raw/", samples[1,2], ".fastq.gz", sep = "")
print(file_output_2)
print(file_input_2)
}

#theoretically would include this in the for loop
#for (x in 1:1) {
#file_output_1 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/", samples[1,1],"_trimmed.fastq.gz", sep = "")
#file_input_1 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_raw/", samples[1,1], ".fastq.gz", sep = "")

#file_output_2 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/", samples[1,2],"_trimmed.fastq.gz", sep = "")
#file_input_2 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_raw/", samples[1,2], ".fastq.gz", sep = "")
#system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o file_output_1 -p file_output_2 file_input_1 file_input_2')


```


#   Brute force cutadapt to get through it
    Obviously would prefer to use an automated way
```{r}
#brute forcing the samples to get it done

#unstimulated 0 minutes
system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep2_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep2_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep2_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep2_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep3_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep3_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep3_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep3_2.fastq.gz')

#siinfekl
system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep1_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep1_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep1_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep1_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep2_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep2_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep2_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep2_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep3_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep3_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep3_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep3_2.fastq.gz')

#siinfekl + drug
system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep1_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep1_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep1_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep1_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep2_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep2_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep2_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep2_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep3_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep3_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep3_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep3_2.fastq.gz')

#30 unstimulated
system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep1_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep1_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep1_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep1_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep2_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep2_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep2_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep2_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep3_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep3_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep3_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep3_2.fastq.gz')


#run fastqc on the trimmed fastq files
fastqcr::fastqc(fq.dir = "/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed",
                qc.dir = "/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed_qc",
                threads = 1,
                fastqc.path = "~/software/FastQC/fastqc")

#multiqc to check qc data
system(command = 'multiqc /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed_qc --filename multiqc_trimmed.fastqc.html --outdir /home/willettb/bulk-rnaseq-gallagher')


```


#   Build index from mm10 reference genome
```{r}
Rsubread::buildindex(basename = "~/bulk-rnaseq-gallagher/reference_data/gapped_rsubread_index", reference = "~/bulk-rnaseq-gallagher/reference_data/mm10.fa", gappedIndex = TRUE, memory = 8000)

```


#   mapping/aligning
    took ~50 minutes to finish one sample. will set up automating code to run overnight
```{r}
Rsubread::align(index = "~/bulk-rnaseq-gallagher/reference_data/gapped_rsubread_index", readfile1 = "~/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep1_1_trimmed.fastq.gz", readfile2 = "~/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep1_2_trimmed.fastq.gz", output_file = "~/bulk-rnaseq-gallagher/bam_files/0_min_unstimulated_rep1_sorted_bam", sortReadsByCoordinates = TRUE, type = "rna", useAnnotation = TRUE, annot.ext = "~/bulk-rnaseq-gallagher/reference_data/mm10.refGene.gtf", isGTF = TRUE, GTF.featureType = "exon", GTF.attrType = "gene_id")

```

#   mapping/aligning automation
    [DOES THIS WORK? UPDATE LATER]
```{r}
align_filenames <- read.table("/home/willettb/bulk-rnaseq-gallagher/align_samples.txt",sep=" ",header=FALSE)

for (x in 1:12) {
file_input_1 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/", align_filenames[x,1], sep = "")
file_input_2 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/", align_filenames[x,2], sep = "")
align_output <- paste("/home/willettb/bulk-rnaseq-gallagher/bam_files/", align_filenames[x,3], sep = "")

print(file_input_1)
print(file_input_2)
print(align_output)

Rsubread::align(index = "~/bulk-rnaseq-gallagher/reference_data/gapped_rsubread_index", readfile1 = file_input_1, readfile2 = file_input_2, output_file = align_output, sortReadsByCoordinates = TRUE, type = "rna", useAnnotation = TRUE, annot.ext = "~/bulk-rnaseq-gallagher/reference_data/mm10.refGene.gtf", isGTF = TRUE, GTF.featureType = "exon", GTF.attrType = "gene_id")

}

```

#   Read quantification
```{r}
counts_data <- Rsubread::featureCounts(files = list.files(path = "~/data/bulk_RNA_seq/bam_files", pattern = "bam$", full.names = TRUE), annot.ext = "~/bulk-rnaseq-gallagher/reference_data/mm10.refGene.gtf", isGTFAnnotationFile = TRUE, GTF.featureType = "exon", countMultiMappingReads = TRUE, strandSpecific = 0, isPairedEnd = TRUE)

#save(counts_data, file = 'gallagher_quantification_results.rdat')

```



#   QC with Picard
```{r}
#0min
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=bam_files/0_min_unstimulated_rep1_sorted.bam O=picard/0_min_unstimulated_rep1_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/0_min_unstimulated_rep2_sorted.bam O=picard/0_min_unstimulated_rep2_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/0_min_unstimulated_rep3_sorted.bam O=picard/0_min_unstimulated_rep3_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

#30min unstim
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/0_min_unstimulated_rep1_sorted.bam O=picard/30_min_unstimulated_rep1_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep2_sorted.bam O=picard/30_min_unstimulated_rep2_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep3_sorted.bam O=picard/30_min_unstimulated_rep3_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

#30min siinfekl
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep1_sorted.bam O=picard/30_min_N4_peptide_rep1_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep2_sorted.bam O=picard/30_min_N4_peptide_rep2_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep3_sorted.bam O=picard/30_min_N4_peptide_rep3_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

#30min siinfekl + drug
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep1_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep1_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep2_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep2_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep3_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep3_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

```


#   Size distribution data
    Not sure how to automate this. Had done things like this on Alpine in the past
```{r}

#0min
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=bam_files/0_min_unstimulated_rep1_sorted.bam O=picard/0_min_unstimulated_rep1_size_dist.txt H=picard/0_min_unstimulated_rep1_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/0_min_unstimulated_rep2_sorted.bam O=picard/0_min_unstimulated_rep2_size_dist.txt H=picard/0_min_unstimulated_rep2_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/0_min_unstimulated_rep3_sorted.bam O=picard/0_min_unstimulated_rep3_size_dist.txt H=picard/0_min_unstimulated_rep3_hist.pdf ASSUME_SORTED=true')

#30min unstim
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep1_sorted.bam O=picard/30_min_unstimulated_rep1_size_dist.txt H=picard/30_min_unstimulated_rep1_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep2_sorted.bam O=picard/30_min_unstimulated_rep2_size_dist.txt H=picard/30_min_unstimulated_rep2_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep3_sorted.bam O=picard/30_min_unstimulated_rep3_size_dist.txt H=picard/30_min_unstimulated_rep3_hist.pdf ASSUME_SORTED=true')

#30min siinfekl
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep1_sorted.bam O=picard/30_min_N4_peptide_rep1_size_dist.txt H=picard/30_min_N4_peptide_rep1_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep2_sorted.bam O=picard/30_min_N4_peptide_rep2_size_dist.txt H=picard/30_min_N4_peptide_rep2_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep3_sorted.bam O=picard/30_min_N4_peptide_rep3_size_dist.txt H=picard/30_min_N4_peptide_rep3_hist.pdf ASSUME_SORTED=true')

#30min siinfekl + drug
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep1_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep1_size_dist.txt H=picard/30_min_N4_peptide_PRN-694_rep1_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep2_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep2_size_dist.txt H=picard/30_min_N4_peptide_PRN-694_rep2_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep3_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep3_size_dist.txt H=picard/30_min_N4_peptide_PRN-694_rep3_hist.pdf ASSUME_SORTED=true')

```


#   Get info on strandedness
```{r}

#can automate later..
#run these in terminal *not* R
#remember the order though for labeling each line with the group and rep number!

sed -n '7,8p' 0_min_unstimulated_rep1_rna_metrics.txt > summary.txt
sed -n '7,8p' 0_min_unstimulated_rep2_rna_metrics.txt >> summary.txt
sed -n '7,8p' 0_min_unstimulated_rep3_rna_metrics.txt >> summary.txt

sed -n '7,8p' 30_min_unstimulated_rep1_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_unstimulated_rep2_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_unstimulated_rep3_rna_metrics.txt >> summary.txt

sed -n '7,8p' 30_min_N4_peptide_rep1_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_N4_peptide_rep2_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_N4_peptide_rep3_rna_metrics.txt >> summary.txt

sed -n '7,8p' 30_min_N4_peptide_PRN-694_rep1_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_N4_peptide_PRN-694_rep2_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_N4_peptide_PRN-694_rep3_rna_metrics.txt >> summary.txt

#data is unstranded

```




#   Create geneCounts matrix and upload metadata
    make sure each file is formatted properly for analysis
```{r}
gallagher_metadata <- read.csv(file = 'gallagher_metadata.csv', header = TRUE, sep = ",")
rownames(gallagher_metadata)
rownames(gallagher_metadata) <- gallagher_metadata$sample_id
rownames(gallagher_metadata)

geneCounts <- counts_data$counts
all(rownames(gallagher_metadata) == colnames(geneCounts))

colnames(geneCounts)
#removing "_sorted.bam" from the column names
colnames(geneCounts) <- stringr::str_remove(string = colnames(geneCounts), pattern = "_sorted.bam")
colnames(geneCounts)

rownames(gallagher_metadata)
colnames(geneCounts)
geneCounts <- geneCounts[,c(1,2,3,10,11,12,7,8,9,4,5,6)]
rownames(gallagher_metadata)
all(rownames(gallagher_metadata) == colnames(geneCounts))

all(rownames(gallagher_metadata) %in% colnames(geneCounts))
all(colnames(geneCounts) %in% rownames(gallagher_metadata))

#metadata row names = geneCounts colnames

```



#   Read quantification
```{r}


```