---
title: "Gallagher et al 2021 replicate analysis"
output: html_notebook
        replicating some data analysis from gallagher et al 2021
---



#   QC: FastQC and MultiQC
```{r}
#FastQC
fastqcr::fastqc(fq.dir = "~/data/bulk_RNA_seq/fastq_files/raw",
                qc.dir = "/home/willettb/bulk-rnaseq-gallagher/fastq_raw_qc",
                threads = 1,
                fastqc.path = "~/software/FastQC/fastqc")
#multiQC
system(command = 'multiqc /home/willettb/bulk-rnaseq-gallagher/fastq_raw_qc --filename *fastqc.html --outdir /home/willettb/bulk-rnaseq-gallagher')


```


#   CutAdapt - 1 sample -> cant use regex because the samples are paired
    Tried to automate this. Some issue with sending the variables from R into the terminal. Not quite sure how to fix that..?

    I know the problem. But don't know how to fix it. the R variables aren't being passed into the console:
    [Errno 2] No such file or directory: 'file_input_1'
    but if I play around with system(paste()) and system(sprintf()) to print out the variable, I get:
    "sh: 1: /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep1_1.fastq.gz: Exec format error"

```{r}
system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep1_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep1_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep1_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep1_2.fastq.gz')

#automation attempt:
samples <- read.table("/home/willettb/bulk-rnaseq-gallagher/samples.txt",sep=" ",header=FALSE)

for (x in 1:3) {
file_output_1 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/", samples[x,1],"_trimmed.fastq.gz", sep = "")
file_input_1 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_raw/", samples[x,1], ".fastq.gz", sep = "")

file_output_2 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/", samples[x,2],"_trimmed.fastq.gz", sep = "")
file_input_2 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_raw/", samples[x,2], ".fastq.gz", sep = "")

#print(file_input_1)
#print(file_input_2)

#print(file_output_1)
#print(file_output_2)

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o file_output_1 -p file_output_2 file_input_1 file_input_2')
}
```

#   Brute forcing CutAdapt
```{r}
#last two 0min unstimulated samples
system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep2_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep2_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep2_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep2_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep3_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep3_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep3_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/0_min_unstimulated_rep3_2.fastq.gz')

#siinfekl
system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep1_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep1_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep1_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep1_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep2_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep2_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep2_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep2_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep3_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_rep3_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep3_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_rep3_2.fastq.gz')

#siinfekl + drug
system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep1_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep1_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep1_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep1_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep2_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep2_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep2_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep2_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep3_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_N4_peptide_PRN-694_rep3_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep3_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_N4_peptide_PRN-694_rep3_2.fastq.gz')

#30 unstim
system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep1_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep1_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep1_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep1_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep2_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep2_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep2_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep2_2.fastq.gz')

system(command = 'cutadapt -a AGATCGGAAGAG -A AGATCGGAAGAG -q 20 -m 10 --pair-filter=any -o /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep3_1_trimmed.fastq.gz -p /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/30_min_unstimulated_rep3_2_trimmed.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep3_1.fastq.gz /home/willettb/bulk-rnaseq-gallagher/fastq_raw/30_min_unstimulated_rep3_2.fastq.gz')



```

#   FastQC/multiQC on trimmed fastq files
```{r}
fastqcr::fastqc(fq.dir = "/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed",
                qc.dir = "/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed_qc",
                threads = 1,
                fastqc.path = "~/software/FastQC/fastqc")

system(command = 'multiqc /home/willettb/bulk-rnaseq-gallagher/fastq_trimmed_qc --filename multiqc_trimmed.fastqc.html --outdir /home/willettb/bulk-rnaseq-gallagher')
```

#   Build index
```{r}
BiocManager::install("Rsubread")

#build index from mm10 reference genome to speed up mapping/aligning
Rsubread::buildindex(basename = "~/bulk-rnaseq-gallagher/reference_data/gapped_rsubread_index", reference = "~/bulk-rnaseq-gallagher/reference_data/mm10.fa", gappedIndex = TRUE, memory = 8000)

```


#   mapping/aligning
```{r}
Rsubread::align(index = "~/bulk-rnaseq-gallagher/reference_data/gapped_rsubread_index", readfile1 = "~/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep1_1_trimmed.fastq.gz", readfile2 = "~/bulk-rnaseq-gallagher/fastq_trimmed/0_min_unstimulated_rep1_2_trimmed.fastq.gz", output_file = "~/bulk-rnaseq-gallagher/bam_files/0_min_unstimulated_rep1_sorted.bam", sortReadsByCoordinates = TRUE, type = "rna", useAnnotation = TRUE, annot.ext = "~/bulk-rnaseq-gallagher/reference_data/mm10.refGene.gtf", isGTF = TRUE, GTF.featureType = "exon", GTF.attrType = "gene_id")

```

#   automate mapping/aligning. will run this overnight since it took ~50 minutes for the first sample to be aligned

```{r}
align_filenames <- read.table("/home/willettb/bulk-rnaseq-gallagher/align_samples.txt",sep=" ",header=FALSE)

for (x in 1:12) {
file_input_1 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/", align_filenames[x,1], sep = "")
file_input_2 <- paste("/home/willettb/bulk-rnaseq-gallagher/fastq_trimmed/", align_filenames[x,2], sep = "")
align_output <- paste("/home/willettb/bulk-rnaseq-gallagher/bam_files/", align_filenames[x,3], sep = "")

#check names are correct
#print(file_input_1)
#print(file_input_2)
#print(align_output)

Rsubread::align(index = "~/bulk-rnaseq-gallagher/reference_data/gapped_rsubread_index", readfile1 = file_input_1, readfile2 = file_input_2, output_file = align_output, sortReadsByCoordinates = TRUE, type = "rna", useAnnotation = TRUE, annot.ext = "~/bulk-rnaseq-gallagher/reference_data/mm10.refGene.gtf", isGTF = TRUE, GTF.featureType = "exon", GTF.attrType = "gene_id")

}


```

#   Read quantification and save
```{r}
#list to get all the correct bam files for read quantification
trial_list <- list.files(path = "~/data/bulk_RNA_seq/bam_files", pattern = "bam$", full.names = TRUE)
print(trial_list)

#Read quantification
counts_data <- Rsubread::featureCounts(files = list.files(path = "~/data/bulk_RNA_seq/bam_files", pattern = "bam$", full.names = TRUE), annot.ext = "~/bulk-rnaseq-gallagher/reference_data/mm10.refGene.gtf", isGTFAnnotationFile = TRUE, GTF.featureType = "exon", countMultiMappingReads = TRUE, strandSpecific = 0, isPairedEnd = TRUE)

save(counts_data, file = 'gallagher_quantification_results.rdat')
```


#   QC of quantification with Picard - again brute force
    important to get wd right for this
    wdir needed: bulk-rnaseq-gallagher
```{r}
#0min
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=bam_files/0_min_unstimulated_rep1_sorted.bam O=picard/0_min_unstimulated_rep1_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/0_min_unstimulated_rep2_sorted.bam O=picard/0_min_unstimulated_rep2_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/0_min_unstimulated_rep3_sorted.bam O=picard/0_min_unstimulated_rep3_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

#30min unstim
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep1_sorted.bam O=picard/30_min_unstimulated_rep1_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep2_sorted.bam O=picard/30_min_unstimulated_rep2_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep3_sorted.bam O=picard/30_min_unstimulated_rep3_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

#30min siinfekl
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep1_sorted.bam O=picard/30_min_N4_peptide_rep1_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep2_sorted.bam O=picard/30_min_N4_peptide_rep2_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep3_sorted.bam O=picard/30_min_N4_peptide_rep3_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

#30min siinfekl + drug
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep1_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep1_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep2_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep2_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectRnaSeqMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep3_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep3_rna_metrics.txt REF_FLAT=reference_data/mm10.refflat RIBOSOMAL_INTERVALS=reference_data/mm10_primary_ribosomalRNA_interval_list.txt STRAND=NONE')



```

#   size distribution data. generates .txt file and histogram image
```{r}

#0min
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=bam_files/0_min_unstimulated_rep1_sorted.bam O=picard/0_min_unstimulated_rep1_size_dist.txt H=picard/0_min_unstimulated_rep1_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/0_min_unstimulated_rep2_sorted.bam O=picard/0_min_unstimulated_rep2_size_dist.txt H=picard/0_min_unstimulated_rep2_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/0_min_unstimulated_rep3_sorted.bam O=picard/0_min_unstimulated_rep3_size_dist.txt H=picard/0_min_unstimulated_rep3_hist.pdf ASSUME_SORTED=true')

#30min unstim
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep1_sorted.bam O=picard/30_min_unstimulated_rep1_size_dist.txt H=picard/30_min_unstimulated_rep1_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep2_sorted.bam O=picard/30_min_unstimulated_rep2_size_dist.txt H=picard/30_min_unstimulated_rep2_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_unstimulated_rep3_sorted.bam O=picard/30_min_unstimulated_rep3_size_dist.txt H=picard/30_min_unstimulated_rep3_hist.pdf ASSUME_SORTED=true')

#30min siinfekl
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep1_sorted.bam O=picard/30_min_N4_peptide_rep1_size_dist.txt H=picard/30_min_N4_peptide_rep1_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep2_sorted.bam O=picard/30_min_N4_peptide_rep2_size_dist.txt H=picard/30_min_N4_peptide_rep2_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_rep3_sorted.bam O=picard/30_min_N4_peptide_rep3_size_dist.txt H=picard/30_min_N4_peptide_rep3_hist.pdf ASSUME_SORTED=true')

#30min siinfekl + drug
system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep1_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep1_size_dist.txt H=picard/30_min_N4_peptide_PRN-694_rep1_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep2_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep2_size_dist.txt H=picard/30_min_N4_peptide_PRN-694_rep2_hist.pdf ASSUME_SORTED=true')

system(command='java -jar ~/software/picard/build/libs/picard.jar CollectInsertSizeMetrics I=../data/bulk_RNA_seq/bam_files/30_min_N4_peptide_PRN-694_rep3_sorted.bam O=picard/30_min_N4_peptide_PRN-694_rep3_size_dist.txt H=picard/30_min_N4_peptide_PRN-694_rep3_hist.pdf ASSUME_SORTED=true')

```

#   get info from rna metrics for strandedness (data is unstranded)
    can automate later..
    run these in terminal *not* R
    remember the order though for labeling each line with the group and rep number!
```{r}
sed -n '7,8p' 0_min_unstimulated_rep1_rna_metrics.txt > summary.txt
sed -n '7,8p' 0_min_unstimulated_rep2_rna_metrics.txt >> summary.txt
sed -n '7,8p' 0_min_unstimulated_rep3_rna_metrics.txt >> summary.txt

sed -n '7,8p' 30_min_unstimulated_rep1_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_unstimulated_rep2_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_unstimulated_rep3_rna_metrics.txt >> summary.txt

sed -n '7,8p' 30_min_N4_peptide_rep1_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_N4_peptide_rep2_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_N4_peptide_rep3_rna_metrics.txt >> summary.txt

sed -n '7,8p' 30_min_N4_peptide_PRN-694_rep1_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_N4_peptide_PRN-694_rep2_rna_metrics.txt >> summary.txt
sed -n '7,8p' 30_min_N4_peptide_PRN-694_rep3_rna_metrics.txt >> summary.txt
```

#   upload labeled metadata (gallagher_metadata.csv)
    filter out genes with <15 reads across all samples
```{r}
gallagher_metadata <- read.csv(file = 'gallagher_metadata.csv', header = TRUE, sep = ",")
rownames(gallagher_metadata)
rownames(gallagher_metadata) <- gallagher_metadata$sample_id
rownames(gallagher_metadata)

geneCounts <- counts_data$counts
all(rownames(gallagher_metadata) == colnames(geneCounts))

colnames(geneCounts)
#removing "_sorted.bam" from the column names
colnames(geneCounts) <- stringr::str_remove(string = colnames(geneCounts), pattern = "_sorted.bam")
colnames(geneCounts)

rownames(gallagher_metadata)
colnames(geneCounts)
geneCounts <- geneCounts[,c(1,2,3,10,11,12,7,8,9,4,5,6)]
rownames(gallagher_metadata)
all(rownames(gallagher_metadata) == colnames(geneCounts))

all(rownames(gallagher_metadata) %in% colnames(geneCounts))
all(colnames(geneCounts) %in% rownames(gallagher_metadata))

#metadata row names = geneCounts colnames

summary(geneCounts)
dim(geneCounts)

#gene filtering
#using 15 reads across all samples
keep <- rowSums(geneCounts) >= 15

filteredGeneCounts <- geneCounts[keep,]
dim(filteredGeneCounts)


```

#   generate scaling factor for DE analysis via DEseq2
```{r}
library(DESeq2)
library(limma)

deObj <- DESeq2::DESeqDataSetFromMatrix(countData = filteredGeneCounts, colData = gallagher_metadata, design = ~ biological_group)

deObj <- DESeq2::DESeq(deObj)

#dataframe of normalized counts
normCountsDF <- DESeq2::counts(deObj, normalized = TRUE)

#principle component analysis
pcs <- prcomp(t(normCountsDF), scale. = TRUE)
summary(pcs)

pc_variance_summary <- summary(pcs)

gallagher_metadata <- cbind(pcs$x, gallagher_metadata)


```

#   Initial PCA analysis
```{r}

library(ggplot2)
#PC1 vs PC2
ggplot(gallagher_metadata, aes(x = PC1, y = PC2, color = biological_group)) + geom_point(size = 4) + xlab(paste("PC1 -", pc_variance_summary$importance["Proportion of Variance", "PC1"] * 100, "%", sep = "")) + ylab(paste("PC2 -", pc_variance_summary$importance["Proportion of Variance", "PC2"] * 100, "%", sep = "")) + theme_minimal()
#issue with PC1

#PC2 vs PC3
ggplot(gallagher_metadata, aes(x = PC2, y = PC3, color = biological_group)) + geom_point(size = 4) + xlab(paste("PC2 -", pc_variance_summary$importance["Proportion of Variance", "PC2"] * 100, "%", sep = "")) + ylab(paste("PC3 -", pc_variance_summary$importance["Proportion of Variance", "PC3"] * 100, "%", sep = "")) + theme_minimal()

#issue is the replicate
ggplot(gallagher_metadata, aes(x = PC1, y = PC2, color = factor(replicate))) + geom_point(size = 4) + xlab(paste("PC1 -", pc_variance_summary$importance["Proportion of Variance", "PC1"] * 100, "%", sep = "")) + ylab(paste("PC2 -", pc_variance_summary$importance["Proportion of Variance", "PC2"] * 100, "%", sep = "")) + theme_minimal()

#maybe ribosomal bases also contributes a little. not as clearly as replicate though.
ggplot(gallagher_metadata, aes(x = PC1, y = PC2, color = PCT_RIBOSOMAL_BASES)) + geom_point(size = 4) + xlab(paste("PC1 -", pc_variance_summary$importance["Proportion of Variance", "PC1"] * 100, "%", sep = "")) + ylab(paste("PC2 -", pc_variance_summary$importance["Proportion of Variance", "PC2"] * 100, "%", sep = "")) + theme_minimal()

```


#   add replicate covariate into DESeq2 data object (make sure its a factor not an integer)
```{r}
deObj$replicate <- factor(deObj$replicate)
deObj@design <- ~factor(replicate) + PCT_RIBOSOMAL_BASES + biological_group

#update DESeq2 data object
deObj <- DESeq2::DESeq(deObj)

#-----
#NOTE: normalized counts matrix has not changed. covariate analysis is not factored into that!
#do NOT run DE analysis on the normalized counts matrix if I needed to alter the covariate analysis!
#-----

vst_de_obj <- DESeq2::vst(deObj, blind=FALSE)
vst_matrix <- assay(vst_de_obj)

model_matrix1 <- model.matrix(~biological_group, colData(vst_de_obj))

vst_matrix <- limma::removeBatchEffect(vst_matrix, batch=vst_de_obj$replicate,covariates=vst_de_obj$PCT_RIBOSOMAL_BASES, design=model_matrix1)
assay(vst_de_obj) <- vst_matrix

#looks good
plotPCA(vst_de_obj, intgroup = c("biological_group"))
plotPCA(vst_de_obj, intgroup = c("replicate"))

#covariates factored in were replicate and PCT_RIBOSOMAL_BASES
#didn't need ribosomal bases, but wanted to make sure the code worked out for my own sake
```

#   Differential gene expression analysis
```{r}
prn_rel_30min_n4 <- DESeq2::results(deObj, contrast = c("biological_group","30_min_N4_PRN-694","30_min_N4"))
prn_rel_30min_n4

prn_rel_30_min_n4_df <- as.data.frame(prn_rel_30min_n4)
write.csv(x=prn_rel_30_min_n4_df, file="drug_vs_untreated_N4_30min.csv", quote = F, row.names = T)

#initial differential expression done between simulated control and treated groups
#save data objects so we don't need to redo the analysis

save(list = c("gallagher_metadata", "deObj", "geneCounts", "pcs", "normCountsDF"), file = "bulk_rnaseq_exp_gallagher2021.Rdat")
```


##  QC for DE analysis
    save
```{r}
boxplot(log10(assays(deObj)[["cooks"]]), range=0, las=1)
hist(prn_rel_30min_n4$pvalue)

write.csv(as.data.frame(prn_rel_30min_n4), "30min_n4_prn_vs_30min_n4.csv", quote = F)
```

##  volcano plot
```{r}
library(ggrepel)
deResults <- read.csv(file = '30min_n4_prn_vs_30min_n4.csv', header = TRUE, row.names = 1)
deResults$negLog10FDR <- (-1)*log10(deResults$padj)
deResults$gene <- rownames(deResults)

ggplot(deResults, aes(x = log2FoldChange, y = negLog10FDR)) + geom_point() + geom_vline(xintercept=c(-1,1), col="darkred") + xlim(-5,5) + geom_hline(yintercept=1, col = "darkred") + theme_minimal() + geom_text_repel(data=subset(deResults, padj <0.05), aes (label = gene), max.overlaps = 15)
```


##  Heatmap
```{r}
library(pheatmap)
sigGenes <- deResults[which(deResults$pad < 0.1), "gene"]
samplesOfInterest <- c("30_min_N4_peptide_rep1", "30_min_N4_peptide_rep2", "30_min_N4_peptide_rep3", "30_min_N4_peptide_PRN-694_rep1", "30_min_N4_peptide_PRN-694_rep2", "30_min_N4_peptide_PRN-694_rep3")

dataToGraph <- normCountsDF[sigGenes, samplesOfInterest]
pallette <- colorRampPalette(c("#6aa84f", "white", "#ffff00"))

pheatmap(dataToGraph, show_rownames = TRUE, scale = 'row', cluster_cols = TRUE, cluster_rows = TRUE, angle_col = 45, cellwidth = 25, cellheight = 10, color = pallette(20), display_number = FALSE)
```

#   OverRepresentation analysis
```{r}
library(msigdbr)
library(org.Mm.eg.db)
library(clusterProfiler)
```

##  Hallmark pathways
    Not sure why the output for the under repped genes isn't showing..?
```{r}
upGenes <- deResults[which(deResults$padj < 0.05 & deResults$log2FoldChange > 0.25), "gene"]
downGenes <- deResults[which(deResults$padj < 0.05 & deResults$log2FoldChange < -0.25), "gene"]

msigH_mouse <- msigdbr(species = "Mus musculus", category = "H") %>% dplyr::select(gs_name, gene_symbol)

outputH_up <- enricher(gene = upGenes, TERM2GENE = msigH_mouse)
outputH_down <- enricher(gene = downGenes, TERM2GENE = msigH_mouse)

hallmark_results_up <- outputH_up@result
hallmark_results_down <- outputH_down@result

#save outputs and data
write.table(x = hallmark_results_up, file = "msigHallmark_up.tsv", quote = FALSE, row.names = FALSE, sep = '\t')
write.table(x = hallmark_results_down, file = "msigHallmark_down.tsv", quote = FALSE, row.names = FALSE, sep = '\t')

save(list = c("outputH_up", "outputH_down"), file = "msigH_results.Rdat")

#visualize
dotplot(outputH_up, showCategory = 1, x="GeneRatio", color = "p.adjust", title="up")
dotplot(outputH_down, showCategory = 7, x="GeneRatio", color = "p.adjust", title="down")

#not sure why outputH_up isn't working. Down is though...
#maybe has something to do with there only being 2 entries in the up dataset??
```


##  C7 immunological gene sets
    
    I'm not sure why dotplot only showed 5 entries. Tcell_pathways_up$ID is a list of ~42 or something like that
    either way I subsetted the C7_results_up into T cell specific pathways that have previously been published.
    Not informative in this study persay, but this would be how I approach this type of analysis in the future:
    first subsetting the different C7 pathways into ones that are relevant for the cell type of interest.
```{r}
msigC7_mouse <- msigdbr(species = "Mus musculus", category = "C7") %>% dplyr::select(gs_name, gene_symbol)

outputC7_up <- enricher(gene = upGenes, TERM2GENE = msigC7_mouse)
outputC7_down <- enricher(gene = downGenes, TERM2GENE = msigC7_mouse)

C7_results_up <- outputC7_up@result
C7_results_down <- outputC7_down@result

dotplot(outputC7_up, showCategory = 10, x="GeneRatio", color = "p.adjust", title="up")

dotplot(outputC7_down, showCategory = 15, x="GeneRatio", color = "p.adjust", title="new", label_format = 40, font.size = 10)

Tcell_pathways_up <- C7_results_up[grepl("CD8_TCELL", C7_results_up$ID),]
dotplot(Tcell_pathways_up, showCategory = 10, x="GeneRatio", color = "p.adjust", title="cd8 Tcell-up")

#this works. I want to substitute these three selected pathways for all "Tcell" pathways that I subsetted earlier
dotplot(outputC7_up, showCategory = c("GSE27786_CD8_TCELL_VS_MONO_MAC_UP","GSE3039_CD4_TCELL_VS_ALPHAALPHA_CD8_TCELL_DN","GSE33424_CD161_INT_VS_NEG_CD8_TCELL_UP"), x="GeneRatio", color = "p.adjust", title="cd8 Tcell-up")

dotplot(outputC7_up, showCategory = test_Tcell$ID, x="GeneRatio", color = "p.adjust", title="cd8 Tcell-up")

dotplot(outputC7_up, showCategory = Tcell_pathways_up$ID, x="GeneRatio", color = "p.adjust", title="cd8 Tcell-up")


```

#   
```{r}

```